src/Service/Resolvers/MsSqlQueryBuilder.cs

```diff
@@ -71,7 +71,7 @@ public string Build(SqlInsertStructure structure)
         public string Build(SqlUpdateStructure structure)
         {
             string predicates = JoinPredicateStrings(
-                                   structure.DbPolicyPredicates,
+                                   structure.DbPolicyPredicatesForOperation[Config.Operation.Update],
```

Comment:
structure.DbPolicyPredicatesForOperation[Config.Operation.Update` will throw when key is not found, how do you ensure this doesn't happen or that an exception is properly handled? 

[docs](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.item?view=net-6.0)

Reply:
Should we use a TryGet here?

Reply:
The key will always be found and the value would be null if there is no db policy specified, see here:
https://github.com/Azure/data-api-builder/blob/39afe04c39a2e28e03b136aafa87a77c1e802596/src/Service/Resolvers/Sql%20Query%20Structures/BaseSqlQueryStructure.cs#L492